src/OAuth2Login.js

```diff
@@ -17,13 +17,15 @@ class OAuth2Login extends Component {
     popupWidth: PropTypes.number,
     redirectUri: PropTypes.string,
     scope: PropTypes.string,
+    state: PropTypes.string,
     disabled: PropTypes.bool
   }
 
   static defaultProps = {
     buttonText: 'Sign in',
     redirectUri: '',
     authorizeUri: '',
+    state: '',
```

Comment:
nit: `state` in React denotes the state of the component, is there a different name we could use? 

It's not really a blocker because this component has no react state (at least for now).

Reply:
To keep it more generic, we could maybe add a `redirectParams` array property, to append a list of parameters to the redirect URI. So in the example it would be: 
```
redirectParams={[
  { state: 'hello' }
]}
```